prime-router/src/main/kotlin/cli/ProcessDataCommands.kt

```diff
@@ -0,0 +1,458 @@
+                    .map { it.first to getOutputFormat(it.second.format) }
+            routeTo != null -> {
+                val pair = translator.translate(
+                    input = inputReport,
+                    toReceiver = routeTo!!,
+                    defaultValues = getDefaultValues()
+                )
+                if (pair != null)
+                    listOf(pair.first to getOutputFormat(pair.second.format))
+                else
+                    emptyList()
+            }
+            outputSchema != null -> {
+                val toSchema = metadata.findSchema(outputSchema!!) ?: error("outputSchema is invalid")
+                val mapping = translator.buildMapping(
+                    toSchema = toSchema,
+                    fromSchema = inputReport.schema,
+                    defaultValues = getDefaultValues()
+                )
+                if (mapping.missing.isNotEmpty()) {
+                    error(
+                        "Error: When translating to $'${toSchema.name} " +
+                            "missing fields for ${mapping.missing.joinToString(", ")}"
+                    )
+                }
+                val toReport = inputReport.applyMapping(mapping)
+                listOf(Pair(toReport, getOutputFormat(Report.Format.CSV)))
+            }
+            else -> listOf(Pair(inputReport, getOutputFormat(Report.Format.CSV)))
+        }
+
+        if (warnings.size > 0) {
+            echo("Problems occurred during translation to output schema:")
+            warnings.forEach {
+                echo("${it.scope} ${it.id}: ${it.details}")
+            }
+            echo()
+        }
+
+        // Output reports
+        writeReportsToFile(outputReports) { report, format, stream ->
+            val hl7Configuration = Hl7Configuration(
+                nameFormat = getNameFormat(Report.NameFormat.STANDARD),
+                suppressQstForAoe = suppressQstForAoe,
+                receivingApplicationName = receivingApplication,
+                receivingFacilityName = receivingFacility,
+                receivingOrganization = receivingOrganization,
+                receivingApplicationOID = "",
+                receivingFacilityOID = "",
+                messageProfileId = "",
+                useBatchHeaders = format == Report.Format.HL7_BATCH,
+                reportingFacilityId = reportingFacilityId,
+                reportingFacilityName = reportingFacilityName,
+            )
+            when (format) {
+                Report.Format.INTERNAL -> csvSerializer.writeInternal(report, stream)
+                Report.Format.CSV -> csvSerializer.write(report, stream)
+                Report.Format.HL7 -> hl7Serializer.write(report, stream, hl7Configuration)
+                Report.Format.HL7_BATCH -> hl7Serializer.writeBatch(report, stream, hl7Configuration)
+                Report.Format.REDOX -> redoxSerializer.write(report, stream)
+            }
+        }
+    }
+}
```

Comment:
I moved the `ProcessData` class into a separate file to start cleaning up `main` and follow the pattern we're doing with other CLI actions.

Reply:
Thank you, thank you, thank you!   

I actually wrote a 'ProcessData.kt' like a month ago, but then on the same day Rick did a zillion changes to the same code in main.kt, so I tossed mine into the bit bucket.  Very glad to see this coming to fruition.